clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -681,7 +736,7 @@ private void startSequencesAtBeginning(TopicPartition topicPartition) {
             log.info("Resetting sequence number of batch with current sequence {} for partition {} to {}",
                     inFlightBatch.baseSequence(), inFlightBatch.topicPartition, sequence.value);
             inFlightBatch.resetProducerState(new ProducerIdAndEpoch(inFlightBatch.producerId(),
-                    inFlightBatch.producerEpoch()), sequence.value, inFlightBatch.isTransactional());
+                    this.producerIdAndEpoch.epoch), sequence.value, inFlightBatch.isTransactional());
```

Comment:
Yes we should, and yes it matters, because this is invoked both after bumping the epoch and after getting a new producer ID in the epoch overflow case